builtins/amp-ad.js

```diff
@@ -243,21 +244,24 @@ export function installAd(win) {
         this.element.appendChild(this.iframe_);
 
         // Triggered by context.noContentAvailable() inside the ad iframe.
-        listen(this.iframe_, 'no-content', () => {
-          // NOTE(erwinm): guard against an iframe firing off no-content twice.
-          // since there is currently no way to `unlisten`.
-          if (this.isDefaultPlaceholder_ && !this.isDefaultPlaceholderSet_) {
-            this.setDefaultPlaceholder_();
-            this.element.appendChild(this.placeholder_);
-            this.element.removeChild(this.iframe_);
-            this.isDefaultPlaceholderSet_ = true;
-          }
-          this.placeholder_.classList.remove('hidden');
+        const unlisten = listen(this.iframe_, 'no-content', () => {
+          this.noContentHandler_();
+          unlisten();
         });
       }
       return loadPromise(this.iframe_);
     }
 
+    noContentHandler_() {
+      if (this.isDefaultFallback_) {
```

Comment:
Let's add a TODO/bug# to check on the performance of this thing. I don't like this relayout.